src/Examine/LuceneEngine/Search/LuceneBooleanOperationBase.cs

```diff
@@ -70,7 +70,10 @@ INestedBooleanOperation INestedBooleanOperation.AndNot(Func<INestedQuery, INeste
         }
 
         public abstract ISearchResults Execute(int maxResults = 500);
+        public abstract ISearchResults Execute(int take, int skip);
         public abstract IOrdering OrderBy(params SortableField[] fields);
         public abstract IOrdering OrderByDescending(params SortableField[] fields);
+
+        public abstract ISearchResults ExecuteWithSkip(int skip, int? take = null);
```

Comment:
Like the other PR this becomes a breaking change so needs to be virtual with NotImplementedException